# Confidence Interval and Hypothesis Test for the Mean

This notebook presents the evaluation of the mean of the dependent variables for the final Portuguese and Mathematics grades (pG3 and mG3). This project will investigate the effects the independent variables have on the final grades. First we will look at how confident we can be that the final grade lies within the expected range for the mean grades at 95% confidence.



We will test if the mean grade in both Maths and Portuguese is equal to 10. The grading system for both subjects is on a 20 point scale, with 20 being the best possible score and 1 the worst. A mean score of 10 might be expected if the scores are evenly spread across the distribution.


In [1]:
# import required packages

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import scipy.stats as stats
import math

## Import the prepared data

In [2]:
df = pd.read_csv("prepared_dataset.csv")

## Construct a 95% confidence interval for the mean of pG3, the final grade for Portuguese, and mG3, the final grade for Mathematics.

To construct a confidence interval, we need the mean and the half-width of the confidence interval. 

#### Portuguese Final Grade pG3



**1: Find the sample mean**

In [3]:
# pG3 Portuguese:
pt_sample_mean = df["pG3"].mean()
pt_sample_mean

12.554054054054054

In [4]:
# mG3 Mathematics:
mat_sample_mean = df["mG3"].mean()
mat_sample_mean

10.464864864864865

The sample mean is 12.55 for the final Portuguese grade pG3, and 10.46 for the final Mathematics grade mG3.

**2. Find the Standard Error, SE**

In [5]:
# mG3 Mathematics:
mat_std = df["mG3"].std()
mat_se = mat_std/ math.sqrt(df["mG3"].count())
mat_se
mat_std

4.608829526277273

In [6]:
# pG3 Portuguese:
pt_std = df["pG3"].std()
pt_se = pt_std / math.sqrt(df["pG3"].count())
pt_se

0.1530332230418159

The standard error is 4.6088 for  the final mathematics grade mG3, and 0.1530 for the final Portuguese grade.

**3. Calculate the Confidence Interval, ci**

First, we calculate the half-width.

For Two-Tailed test at a 95% confidence level, we want to fence-off 2.5% of the distribution in both of the tails, as measured from the mean.

To use the ppf function, we need the cdf value. At 95% confidence, the cdf value is
 ${{100} - {({{100 - LOC}\over {2}})}\over 100} = 0.975 $.




In [7]:
level_of_confidence = 95

cdf_val = (100-((100-95)/2))/100
cdf_val


0.975

In [8]:
# Portuguese pG3:
pt_ci_halfwidth = stats.norm.ppf(cdf_val, 0, pt_se)

# Mathematics mG3:
mat_ci_halfwidth = stats.norm.ppf(cdf_val, 0, mat_se)

print("At 95% level of confidence: ")
print("Portuguese pG3 Confidence Interval: ", round(pt_sample_mean, 3), " ± ", round(pt_ci_halfwidth, 3))
print("Mathematics mG3 Confidence Interval: ", round(mat_sample_mean, 3), " ± ", round(mat_ci_halfwidth, 3))


At 95% level of confidence: 
Portuguese pG3 Confidence Interval:  12.554  ±  0.3
Mathematics mG3 Confidence Interval:  10.465  ±  0.47


At 95% level of confidence: 

Portuguese final grade pG3 Confidence Interval:  12.554  ±  0.3

Mathematics final grade mG3 Confidence Interval:  10.465  ±  0.47

## Test the hypothesis that the mean is equal to 10

**Statement of the hypothesis for pG3 and mG3**

|**pG3 Portuguese**  | **mG3 Mathematics**|
|--|--|
|H<sub>0</sub>: s<sub>avg</sub> = 10|H<sub>0</sub>: s<sub>avg</sub> = 10|
|H<sub>a</sub>: s<sub>avg</sub> &ne; 10|  H<sub>a</sub>: s<sub>avg</sub> &ne; 10|




This is a two-tail test, for which we must find the **critical values C<sub>1</sub> and C<sub>2</sub>** , for both of our dependent variables the final mathematics and portuguese grades (mG3 and pG3).

In [9]:
#known information
mu = 10
alpha = 0.05

In [10]:
#pG3 Critical Values
pt_C1 = stats.norm.ppf(q=(alpha/2), loc=mu, scale=pt_se)
pt_C2 = stats.norm.ppf(q=(1-alpha/2), loc=mu, scale= pt_se)

print("pG3 C1: ", pt_C1)
print("pG3 C2: ", pt_C2)

pG3 C1:  9.700060394399955
pG3 C2:  10.299939605600045


In [11]:
#mG3 Critical Values
mat_C1 = stats.norm.ppf(q=(alpha/2), loc=mu, scale= mat_se)
mat_C2 = stats.norm.ppf(q=(1-alpha/2), loc=mu, scale= mat_se)

print("mG3 C1: ", mat_C1)
print("mG3 C2: ", mat_C2)

mG3 C1:  9.530389419133678
mG3 C2:  10.469610580866322


**Comparison between mean and critical values**

We check if the hypothesised mean falls within the critical values denoting the 95% range.
Critical_value_1 < sample_mean < Critical_value_2

The null hypothesis is rejected if the mean falls outside the 95% range. 

In [12]:
#check if our sample mean falls between the critical values 
def test_mean_against_CVs(C1, C2, sample_mean):
    print(C1, sample_mean , C2)
    if(C1 < sample_mean < C2):
       
        print("Fail to reject null hypothesis.")
    else: 
        
        print("Reject null hypothesis.")

#Mathematics mG3:
print("Mathematics mG3: ")
test_mean_against_CVs(mat_C1, mat_C2, mat_sample_mean)

print()
#Portuguese pG3
print("Portuguese pG3: ")
test_mean_against_CVs(pt_C1, pt_C2, pt_sample_mean)


Mathematics mG3: 
9.530389419133678 10.464864864864865 10.469610580866322
Fail to reject null hypothesis.

Portuguese pG3: 
9.700060394399955 12.554054054054054 10.299939605600045
Reject null hypothesis.


**Test result interpretation**

The average mean of  the final Portuguese grade pG3 is **not** within the 95% range around the hypothesised mean of 10. Therefore we reject the null hypothesis $H_0$ for pG3. The result is significant at 0.05. The average grade of pG3 is significantly different from 10 at 12.55. 

The average mean of  the final mathematics grade mG3 is within the 95% around its hypothesised mean of 10. Therefore we fail to reject the null hypothesis $H_0$ for mG3. The result for mG3 is not significant at the 0.05 significance level. The average grade of mG3 is not significantly different from 10 at 10.46.


# Conclusion
As the average final Portuguese grade (M = 12.55) falls outside the 95% confidence interval around 10,  it means that the final portuguese grade is higher than would be expected. This suggests that the students perform better in Portuguese class, or the grades are skewed somewhat.

The average final Mathematics grade (M = 10.46) falls within the 95% confidence interval around 10. This shows that the average grade for Mathematics is quite close to the middle of the grading scheme from 0 to 20. 
